File: packages/python-sdk/strait/composition/_idempotency.py
"""Idempotency key header injection."""

from __future__ import annotations


def with_idempotency(headers: dict[str, str] | None, key: str) -> dict[str, str]:
    return with_idempotency_header(headers, key, "Idempotency-Key")


def with_idempotency_header(
    headers: dict[str, str] | None, key: str, header_name: str,
) -> dict[str, str]:
    result = dict(headers) if headers else {}
    result[header_name] = key
    return result

File: packages/go-sdk/composition/idempotency.go
package composition

import "maps"

// WithIdempotency returns a new header map with the idempotency key added.
// If headers is nil, a new map is created.
func WithIdempotency(headers map[string]string, key string) map[string]string {
	return WithIdempotencyHeader(headers, key, "Idempotency-Key")
}

// WithIdempotencyHeader returns a new header map with the idempotency key
// using a custom header name.
func WithIdempotencyHeader(headers map[string]string, key string, headerName string) map[string]string {
	result := make(map[string]string, len(headers)+1)
	maps.Copy(result, headers)
	result[headerName] = key
	return result
}
